### PR TITLE
cluster test

### PR DESCRIPTION
**Pull request by longbowlu**
A test that is run against the entire cluster. e.g.
```
cargo run --bin sui-cluster-test -- staging
cargo run --bin sui-cluster-test -- prod
cargo run --bin sui-cluster-test -- custom --gateway-address {} --faucet-address {}
```
We can run this test as a part of the release process to capture failures immediately after deploying. 

What is covered:
* request coin from faucet service
* transfer coin
* split and merge coin
* interact with move contract

What is NOT covered:
* wallet cli
* explorer

### CODE REVIEW

**Comment by longbowlu** [comment]
> This should probably in the sui/tests directory so that cargo knows its a test.

I'm hesitant to put it there because this "e2e" test case tests against a deployed cluster, instead of a network spinning out locally. So it's probably not sth we want to run in every `cargo test`

> I think some of these tests are duplicates of things with do in cli_tests.rs - or am I missing something?

Their purposes are different (as mentioned above). cli_tests.rs tests the cli tools and this tests the cluster.

**Comment by bmwill** [comment]
I think we should sync. I put out some utilities for making it easier to write e2e tests here https://github.com/MystenLabs/sui/pull/2337 that may simplify some of the work of writing these types of tests.

**Comment by longbowlu** [comment]
> Maybe we don't want to call this `e2e-test` then? maybe `cluster-health-check` or something to indicate its not intended as a "test" in the CI sense of the word.

Rename all "e2e" to "cluster" to avoid confusion. Thanks for the advice

File: Cargo.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ members = [
     "crates/sui-open-rpc",
     "crates/sui-open-rpc-macros",
     "crates/sui-storage",
+    "crates/sui-testsuite",
```
Comment (bmwill) [inline Cargo.toml:20]: lets not call this a testsuite as we probably want to reserve that for an actual testsuite as this seems to be more about checking a cluster's health than an actual suite of tests to run at ci time.

Comment (longbowlu) [reply Cargo.toml:20]: I'm actually thinking about extending this into a comprehensive CI suite. I will read your PR and maybe sync offline on how to move forward. 
For now I will rename it to "sui-cluster-test"

**Review by tharbert — APPROVED** [review]
This is great. How are you expecting it to be used? Can I simply build and run 'cargo run --bin sui-e2e-test -- $env' at the end of a deploy? What about periodically executing it for active monitoring?

**Comment by mystenmark** [comment]
> > This should probably in the sui/tests directory so that cargo knows its a test.
> 
> I'm hesitant to put it there because this "e2e" test case tests against a deployed cluster, instead of a network spun out locally. So it's probably not sth we want to run in every `cargo test`
> 
> > I think some of these tests are duplicates of things with do in cli_tests.rs - or am I missing something?
> 
> Their purposes are different (as mentioned above). cli_tests.rs tests the cli tools and this tests the cluster.

Ok, I think I missed the part in the code where this is actually connecting to staging! I withdraw my comments

**Comment by bmwill** [comment]
> > > This should probably in the sui/tests directory so that cargo knows its a test.
> > 
> > 
> > I'm hesitant to put it there because this "e2e" test case tests against a deployed cluster, instead of a network spun out locally. So it's probably not sth we want to run in every `cargo test`
> > > I think some of these tests are duplicates of things with do in cli_tests.rs - or am I missing something?
> > 
> > 
> > Their purposes are different (as mentioned above). cli_tests.rs tests the cli tools and this tests the cluster.
> 
> Ok, I think I missed the part in the code where this is actually connecting to staging! I withdraw my comments

This is why i don't think the name `test` should be included in the name of this tool

**Comment by longbowlu** [comment]
> How are you expecting it to be used? Can I simply build and run 'cargo run --bin sui-e2e-test -- $env' at the end of a deploy? 

yeah I think so, we just copy this binary to the container and run the command 


> What about periodically executing it for active monitoring?

Oh I'd love to have that through, I guess, one of these https://betterprogramming.pub/cron-job-patterns-in-aws-126fbf54a276

**Pull request merged**